### DISCUSSION TITLE
Misleading recommendation for permissions when using compiled binary

### DISCUSSION THREAD

**Discussion by Almenon** [denoland/deno#18728] [General]
When I compile a program like the below

```ts
const file = await Deno.open("C:\\dev\\misc\\candel\\a.ts");
await file.readable.pipeTo(Deno.stdout.writable, { preventClose: true });
```

 and run it (`./cat.exe`) I get the following:
```
┌ ⚠️  Deno requests read access to "C:\dev\misc\candel\a.ts".
├ Requested by `Deno.open()` API.
├ Run again with --allow-read to bypass this prompt.
└ Allow? [y/n/A] (y = yes, allow; n = no, deny; A = allow all read permissions) >
```

However, specifying `./cat.exe --allow-read` results in the same prompt. 

Is this a bug?

**Comment by yooneskh** [comment]
No you need to specify all of the permissions that your script needs in the compile time.

`deno compile --allow-read script.ts`

**Comment by mmastrac** [comment]
I'd consider this a bug -- we should change that message to properly reflect that the permissions should be passed to deno compile. Could you file an issue for this?

**Reply by Almenon** [reply]
done :)
